frontend/src/app/services/eta.service.ts

```diff
@@ -143,7 +145,28 @@ export class EtaService {
 
     if (!isAccelerated) {
       const blocks = mempoolPosition.block + 1;
-      const wait = da.adjustedTimeAvg * (mempoolPosition.block + 1);
+
+      // Estimate future incoming tx rate from mempool statistics
+      let vsizePerSecond = Math.min(
+        this.estimateVsizePerSecond(tx, mempoolStats),
+        0.95 * this.stateService.blockVSize / da.adjustedTimeAvg * 1000
+      );
+
+      // Count the number of blocks until we expect this tx to be mined
+      let blocksUntilMined = 0;
+      let mined = false;
+      let vsize = mempoolPosition.vsize + this.stateService.blockVSize * mempoolPosition.block;
+      // This loop will always terminate because we cap vsizePerSecond to 0.95 * maxCapacity
+      while (!mined) {
+        vsize = vsize + vsizePerSecond * da.adjustedTimeAvg / 1000 - this.stateService.blockVSize;
+        if (vsize + tx.weight / 8 < 0) { // Means that our tx fits in expected next block
+          mined = true;
+        }
+        blocksUntilMined++;
+      }
+
+      const wait = blocksUntilMined * da.adjustedTimeAvg;
```

Comment:
if possible, it would be nice to use an analytic solution for this.

maybe something like:
```typescript
let blocksUntilMined = Infinity;
const vsizeAhead = mempoolPosition.vsize + (this.stateService.blockVSize * mempoolPosition.block);
const incomingVsizePerBlock = vsizePerSecond * da.adjustedTimeAvg / 1000;
const vsizeConsumedPerBlock = incomingVsizePerBlock - this.stateService.blockVSize;
if (vsizeConsumedPerBlock > 0) {
  blocksUntilMined = Math.ceil(vsizeAhead / vsizeConsumedPerBlock);
}
```

Reply:
Yes, that's better than the loop for sure 😁